$NetBSD: patch-ab,v 1.4 2011/12/01 04:05:41 dholland Exp $

Caution: the extra cpp checking (the three hunks containing "Should be
one line") is manually hacked and not reflected in configure.in; do
not regenerate configure with autoconf without merging it. This
checking is required because somewhere between gcc 4.1 and 4.5 cpp
started doing regrettable things with backslash-escaped newlines.

--- configure.orig	2000-06-12 20:05:56.000000000 +0000
+++ configure
@@ -20,6 +20,8 @@ ac_help="$ac_help
 ac_help="$ac_help
   --with-x-toolkit=KIT    use an X toolkit (KIT = yes/lucid/athena/motif/no)"
 ac_help="$ac_help
+  --with-ipv6             support IPv6 stack"
+ac_help="$ac_help
   --with-x                use the X Window System"
 
 # Initialize some variables set by options.
@@ -584,6 +586,12 @@ Currently, \`yes', \`athena' and \`lucid
 
 fi
 
+# Check whether --with-ipv6 or --without-ipv6 was given.
+if test "${with_ipv6+set}" = set; then
+  withval="$with_ipv6"
+  :
+fi
+
 
 #### Make srcdir absolute, if it isn't already.  It's important to
 #### avoid running the path through pwd unnecessary, since pwd can
@@ -718,9 +726,14 @@ case "${canonical}" in
 			machine=hp9000s300 ;;
       powerpc-apple-netbsd*) machine=macppc ;;
       mips-*-netbsd*)	machine=pmax ;;
+      mipsel-*-netbsd*)	machine=pmax ;;
+      mipsev-*-netbsd*)	machine=mips ;;		# HF: what else?
       ns32k-*-netbsd*)	machine=ns32000 ;;
-      sparc-*-netbsd*)	machine=sparc ;;
+      powerpc-*-netbsd*) machine=macppc ;;
+      sparc*-*-netbsd*)	machine=sparc ;;
       vax-*-netbsd*)	machine=vax ;;
+      arm*-*-netbsd*)	machine=arm ;;
+      x86_64-*-netbsd*)	machine=amd64 ;;
     esac
   ;;
 
@@ -1519,6 +1532,7 @@ case "${canonical}" in
       *-sysv4.2uw* )	  	opsys=unixware; NON_GNU_CPP=/lib/cpp ;;
       *-386bsd* )	        opsys=386bsd ;;
       *-freebsd* )	        opsys=freebsd ;;
+      *-dragonfly* )		opsys=dragonfly ;;
       *-nextstep* )             opsys=nextstep ;;
       ## Otherwise, we'll fall through to the generic opsys code at the bottom.
     esac
@@ -1916,15 +1930,21 @@ else
   CPP="${CC-cc} -E"
   # On the NeXT, cc -E runs the code through the compiler's parser,
   # not just through cpp.
+  # And, gcc 4.5? and up does not preserve escaped newlines, which is
+  # fatal for preprocessing makefiles.
   cat > conftest.$ac_ext <<EOF
 #line 1921 "configure"
 #include "confdefs.h"
 #include <assert.h>
 Syntax Error
+Should \\
+be one line.
 EOF
-ac_try="$ac_cpp conftest.$ac_ext >/dev/null 2>conftest.out"
+ac_try="$ac_cpp conftest.$ac_ext >conftest2.out 2>conftest.out"
 { (eval echo configure:1927: \"$ac_try\") 1>&5; (eval $ac_try) 2>&5; }
-ac_err=`grep -v '^ *+' conftest.out | grep -v "^conftest.${ac_ext}\$"`
+ac_err=`grep -v '^ *+' conftest.out | grep -v "^conftest.${ac_ext}\$"; \
+    grep -L "Should  *be one line" conftest2.out \
+    `
 if test -z "$ac_err"; then
   :
 else
@@ -1938,10 +1958,14 @@ else
 #include "confdefs.h"
 #include <assert.h>
 Syntax Error
+Should \\
+be one line.
 EOF
-ac_try="$ac_cpp conftest.$ac_ext >/dev/null 2>conftest.out"
+ac_try="$ac_cpp conftest.$ac_ext >conftest2.out 2>conftest.out"
 { (eval echo configure:1944: \"$ac_try\") 1>&5; (eval $ac_try) 2>&5; }
-ac_err=`grep -v '^ *+' conftest.out | grep -v "^conftest.${ac_ext}\$"`
+ac_err=`grep -v '^ *+' conftest.out | grep -v "^conftest.${ac_ext}\$"; \
+    grep -L "Should  *be one line" conftest2.out \
+    `
 if test -z "$ac_err"; then
   :
 else
@@ -1955,10 +1979,14 @@ else
 #include "confdefs.h"
 #include <assert.h>
 Syntax Error
+Should \\
+be one line.
 EOF
-ac_try="$ac_cpp conftest.$ac_ext >/dev/null 2>conftest.out"
+ac_try="$ac_cpp conftest.$ac_ext >conftest2.out 2>conftest.out"
 { (eval echo configure:1961: \"$ac_try\") 1>&5; (eval $ac_try) 2>&5; }
-ac_err=`grep -v '^ *+' conftest.out | grep -v "^conftest.${ac_ext}\$"`
+ac_err=`grep -v '^ *+' conftest.out | grep -v "^conftest.${ac_ext}\$"; \
+    grep -L "Should  *be one line" conftest2.out \
+    `
 if test -z "$ac_err"; then
   :
 else
@@ -3823,6 +3851,53 @@ else
   echo "$ac_t""no" 1>&6
 fi
 
+  if test x$HAVE_X11XTR6 = xyes; then
+    LIBS="$LIBS -lXpm"
+    echo $ac_n "checking for XawPixmapsInitialize in -lXaw""... $ac_c" 1>&6
+echo "configure:3842: checking for XawPixmapsInitialize in -lXaw" >&5
+ac_lib_var=`echo Xaw'_'XawPixmapsInitialize | sed 'y%./+-%__p_%'`
+if eval "test \"`echo '$''{'ac_cv_lib_$ac_lib_var'+set}'`\" = set"; then
+  echo $ac_n "(cached) $ac_c" 1>&6
+else
+  ac_save_LIBS="$LIBS"
+LIBS="-lXaw  $LIBS"
+cat > conftest.$ac_ext <<EOF
+#line 3850 "configure"
+#include "confdefs.h"
+/* Override any gcc2 internal prototype to avoid an error.  */
+/* We use char because int might match the return type of a gcc2
+    builtin and then its argument prototype would still apply.  */
+char XawPixmapsInitialize();
+
+int main() {
+XawPixmapsInitialize()
+; return 0; }
+EOF
+if { (eval echo configure:3861: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
+  rm -rf conftest*
+  eval "ac_cv_lib_$ac_lib_var=yes"
+else
+  echo "configure: failed program was:" >&5
+  cat conftest.$ac_ext >&5
+  rm -rf conftest*
+  eval "ac_cv_lib_$ac_lib_var=no"
+fi
+rm -f conftest*
+LIBS="$ac_save_LIBS"
+
+fi
+if eval "test \"`echo '$ac_cv_lib_'$ac_lib_var`\" = yes"; then
+  echo "$ac_t""yes" 1>&6
+   LIBXPM="-lXpm" 
+else
+  echo "$ac_t""no" 1>&6
+ LIBXPM=""
+fi
+
+  else
+    LIBXPM=""
+  fi
+  
   LIBS="$OLDLIBS"
 fi
 
@@ -4906,6 +4981,64 @@ EOF
 
 fi
 
+if test "x$with_ipv6" = xyes; then
+  for ac_func in getaddrinfo
+do
+echo $ac_n "checking for $ac_func""... $ac_c" 1>&6
+echo "configure:4973: checking for $ac_func" >&5
+if eval "test \"`echo '$''{'ac_cv_func_$ac_func'+set}'`\" = set"; then
+  echo $ac_n "(cached) $ac_c" 1>&6
+else
+  cat > conftest.$ac_ext <<EOF
+#line 4978 "configure"
+#include "confdefs.h"
+/* System header to define __stub macros and hopefully few prototypes,
+    which can conflict with char $ac_func(); below.  */
+#include <assert.h>
+/* Override any gcc2 internal prototype to avoid an error.  */
+/* We use char because int might match the return type of a gcc2
+    builtin and then its argument prototype would still apply.  */
+char $ac_func();
+
+int main() {
+
+/* The GNU C library defines this for functions which it implements
+    to always fail with ENOSYS.  Some functions are actually named
+    something starting with __ and the normal name is an alias.  */
+#if defined (__stub_$ac_func) || defined (__stub___$ac_func)
+choke me
+#else
+$ac_func();
+#endif
+
+; return 0; }
+EOF
+if { (eval echo configure:5001: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
+  rm -rf conftest*
+  eval "ac_cv_func_$ac_func=yes"
+else
+  echo "configure: failed program was:" >&5
+  cat conftest.$ac_ext >&5
+  rm -rf conftest*
+  eval "ac_cv_func_$ac_func=no"
+fi
+rm -f conftest*
+fi
+
+if eval "test \"`echo '$ac_cv_func_'$ac_func`\" = yes"; then
+  echo "$ac_t""yes" 1>&6
+    ac_tr_func=HAVE_`echo $ac_func | tr 'abcdefghijklmnopqrstuvwxyz' 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'`
+  cat >> confdefs.h <<EOF
+#define $ac_tr_func 1
+EOF
+ 
+else
+  echo "$ac_t""no" 1>&6
+fi
+done
+
+fi
+
 # Set up the CFLAGS for real compilation, so we can substitute it.
 CFLAGS="$REAL_CFLAGS"
 
@@ -5211,6 +5346,7 @@ s%@INSTALL_SCRIPT@%$INSTALL_SCRIPT%g
 s%@INSTALL_DATA@%$INSTALL_DATA%g
 s%@YACC@%$YACC%g
 s%@SET_MAKE@%$SET_MAKE%g
+s%@LIBXPM@%$LIBXPM%g
 s%@ALLOCA@%$ALLOCA%g
 s%@version@%$version%g
 s%@configuration@%$configuration%g
